Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1105 @@
+#include "WarpX.H"
+#include <AMReX_Scan.H>
+#include <AMReX_iMultiFab.H>
+#include <AMReX_MultiFab.H>
+
+/**
+* \brief auxiliary function to count the amount of faces which still need to be extended
+*/
+amrex::Array1D<int, 0, 2>
+WarpX::CountExtFaces() {
+    amrex::Array1D<int, 0, 2> sums{0, 0, 0};
+#ifdef AMREX_USE_EB
+
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        amrex::ReduceOps<amrex::ReduceOpSum> reduce_ops;
+        amrex::ReduceData<amrex::Real> reduce_data(reduce_ops);
+        for (amrex::MFIter mfi(*m_flag_ext_face[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+            amrex::Box const &box = mfi.validbox();
+            auto const &flag_ext_face = m_flag_ext_face[maxLevel()][0]->array(mfi);
```

Comment:
I just noticed that in https://github.com/ECP-WarpX/WarpX/pull/1923/commits/c488941a8cf9c988025e17258ef3092004a81104 this line was changed by replacing `m_flag_ext_face[maxLevel()][idim]->array(mfi)` with `m_flag_ext_face[maxLevel()][0]->array(mfi)` (so replacing `idim` with `0`). Just double checking: was this on purpose?

Reply:
It was not on purpose, thanks a lot for noticing it. I will change it